### PR TITLE
DS transformers for 6.4.0 and 6.3.0

### PR DESCRIPTION
**Pull request by kabir**
https://issues.jboss.org/browse/WFLY-4903

Replaces https://github.com/wildfly/wildfly/pull/7822

### CODE REVIEW

File: connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceDefinition.java [hunk 1]
```diff
@@ -196,6 +196,19 @@ protected boolean isValueDiscardable(PathAddress address, String attributeName,
...
+        builder.getAttributeBuilder()
+                .setDiscard(new DiscardAttributeChecker.DefaultDiscardAttributeChecker() {
+                    @Override
+                    protected boolean isValueDiscardable(PathAddress address, String attributeName, ModelNode attributeValue, TransformationContext context) {
+                        return attributeValue.equals(new ModelNode(false));
+                    }
+                }, TRACKING)
+                .addRejectCheck(RejectAttributeChecker.SIMPLE_EXPRESSIONS, ENABLED)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, TRACKING).end();
+    }
```
Comment (bstansberry) [inline connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceDefinition.java]: This doesn't need the funky stuff related to statistics-enabled that's there for xa-data-source?

Comment (kabir) [inline connector/src/main/java/org/jboss/as/connector/subsystems/datasources/DataSourceDefinition.java]: @bstansberry I would say it is the other way round, the statistics-enabled stuff should be removed from the xa variant

**Comment by bstansberry** [comment]
retest this please

org.jboss.as.test.integration.ejb.management.deployments.EjbInvocationStatisticsTestCase timed out

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
Retest this please

**Pull request merged**